src/main/scala/mesosphere/mesos/TaskBuilder.scala

```diff
@@ -79,7 +80,13 @@ class TaskBuilder(app: AppDefinition,
         val portMappings = c.docker.map { d =>
           d.portMappings.map { pms =>
             pms zip ports map {
-              case (mapping, port) => mapping.copy(hostPort = port.toInt)
+              case (mapping, port) => {
+                if(mapping.containerPort == 0) {
```

Comment:
It would be nice to add a comment here with the reasoning why one would want to set the container port dynamically to the same thing as the host port.